tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/BlockingMiddleware.cs

```diff
@@ -0,0 +1,130 @@
+﻿// <copyright file="BlockingMiddleware.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+#if !NETFRAMEWORK
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Coordinator;
+using Microsoft.AspNetCore.Http;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore;
+
+internal class BlockingMiddleware
+{
+    private readonly bool _endPipeline;
+    private readonly RequestDelegate? _next;
+
+    internal BlockingMiddleware(RequestDelegate? next = null, bool endPipeline = false)
+    {
+        _next = next;
+        _endPipeline = endPipeline;
+    }
+
+    private static Task WriteResponse(HttpContext context, SecuritySettings settings, out bool endedResponse)
+    {
+        var httpResponse = context.Response;
+        if (!httpResponse.HasStarted)
+        {
+            httpResponse.Clear();
+            foreach (var cookie in context.Request.Cookies)
+            {
+                httpResponse.Cookies.Delete(cookie.Key);
+            }
+
+            httpResponse.Headers.Clear();
+            httpResponse.StatusCode = 403;
+            var template = settings.BlockedJsonTemplate;
+            httpResponse.ContentType = "application/json";
+
+            foreach (var header in context.Request.Headers)
+            {
+                if (header.Key == "Accept")
```

Comment:
headers are technically case insensitive, so should probably use 
```suggestion
                if (string.Equals(header.Key, "Accept", StringComparison.OrdinalIgnoreCase))
```